File: server/src/Luyenthi.Domain/Matrixs/Chapter.cs
using Luyenthi.Domain.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Luyenthi.Domain
{
    public class Chapter : IEntity<Guid>, IBaseEntity
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string OrderNumber { get; set; }
        public Guid SubjectId { get; set; }
        public Subject Subject { get; set; }
        public Guid GradeId { get; set; }
        public Grade Grade { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public Guid? CreatedBy { get; set; }
        public Guid? UpdatedBy { get; set; }
        public virtual List<Question> Questions { get; set; }
        public virtual List<Unit> Units { get; set; }

    }
}

File: client/src/models/matrix/Chapter.ts
import { Grade } from "./Grade";
import { Subject } from "./Subject";
import { Unit } from "./Unit";

export interface Chapter {
  name: string;
  id: string;
  subjectId: string;
  gradeId: string;
}
export interface ChapterDetail {
  name: string;
  id: string;
  subjectId: string;
  gradeId: string;
  units: Unit[];
}
export interface ChapterCreate {
  name: string;
  subjectId: string;
  gradeId: string;
}
export interface ChapterDetail {
  id: string;
  name: string;
  subject: Subject;
  grade: Grade;
}
